packages/mui-material/src/Alert/Alert.d.ts

```diff
@@ -11,6 +11,8 @@ export interface AlertPropsVariantOverrides {}
 
 export interface AlertPropsColorOverrides {}
 
+export interface AlertComponentsPropsOverrides {}
```

Comment:
@jake-collibra Is this interface necessary for your use case?

We haven't discussed `*ComponentsPropsOverrides` before so if it is possible, I don't want to include it in this PR just yet.
 
```suggestion
```

Reply:
I believe the intention was to use this in combination with `SlotComponentProps` as done in MUI Base - https://github.com/mui/material-ui/blob/master/packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.types.ts#L32

Reply:
What's the usage look like?

Given:

```ts
{
  componentsProps?: {
    closeButton?: IconButtonProps;
    closeIcon?: SvgIconProps;
  } & AlertComponentsPropsOverrides;
}
```

It looks like the `AlertComponentsPropsOverrides` can add custom slot but I don't think we provide a way to do that at the moment, right?